pkg/gzip/gzip.go

```diff
@@ -69,7 +77,29 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 			)
 		}
 
+		existing := firstExistingParent(fpath)
+		if realParent, err := filepath.EvalSymlinks(existing); err == nil {
+			if !strings.HasPrefix(realParent, realDest) {
```

Comment:
I made the tests evaluate the symlinks in the `tmpdir` paths, so now the tests seem good enough to verify that everything is good.